From 169b01871c264560bb93ca49a41e62d3681c1ad9 Mon Sep 17 00:00:00 2001
From: Amador Pahim <apahim@redhat.com>
Date: Mon, 28 Apr 2014 10:34:13 -0300
Subject: [PATCH 12/14] agent: report Linux cached/buffers memory

Some applications, like Oracle Directory Server, use cached memory
and never release it. Users need to take this memory into account
when planning the memory requirements.

Engine counterpart tracked by bz#1024010.

Change-Id: Id46f85b94937779ea899b05dd1bfa73780a1a4e9
Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=1024007
Signed-off-by: Amador Pahim <apahim@redhat.com>
---
 ovirt-guest-agent/GuestAgentLinux2.py |    2 ++
 tests/message_validator.py            |    2 ++
 2 files changed, 4 insertions(+)

diff --git a/ovirt-guest-agent/GuestAgentLinux2.py b/ovirt-guest-agent/GuestAgentLinux2.py
index 6b4aeac..93ec191 100644
--- a/ovirt-guest-agent/GuestAgentLinux2.py
+++ b/ovirt-guest-agent/GuestAgentLinux2.py
@@ -299,6 +299,8 @@ class LinuxDataRetriver(DataRetriverBase):
         self.memStats['mem_total'] = fields['MemTotal:']
         self.memStats['mem_unused'] = fields['MemFree:']
         self.memStats['mem_free'] = free
+        self.memStats['mem_buffers'] = fields['Buffers:']
+        self.memStats['mem_cached'] = fields['Cached:']
         swap_used = fields['SwapTotal:'] - fields['SwapFree:']
         self.memStats['swap_usage'] = swap_used
         self.memStats['swap_total'] = fields['SwapTotal:']
diff --git a/tests/message_validator.py b/tests/message_validator.py
index 703c453..4cc751a 100644
--- a/tests/message_validator.py
+++ b/tests/message_validator.py
@@ -141,6 +141,8 @@ def validate_memory_stats(msg):
     assert_integral_param(mem, 'mem_free')
     assert_integral_param(mem, 'mem_total')
     assert_integral_param(mem, 'mem_unused')
+    assert_integral_param(mem, 'mem_buffers')
+    assert_integral_param(mem, 'mem_cached')
     assert_integral_param(mem, 'pageflt')
     assert_integral_param(mem, 'swap_in')
     assert_integral_param(mem, 'swap_out')
-- 
1.7.10.4

